### ISSUE TITLE
Variable `tx.ua_hash` and `IP` collection set but not used in CRS core

### ISSUE THREAD

**Issue by M4tteoP** [coreruleset/coreruleset#3068]
Hi there!
`tx.ua_hash`, initialized in [901318](https://github.com/coreruleset/coreruleset/blob/v4.0/dev/rules/REQUEST-901-INITIALIZATION.conf#L295), is just collected in [901321](https://github.com/coreruleset/coreruleset/blob/v4.0/dev/rules/REQUEST-901-INITIALIZATION.conf#L305) inside the `IP` collection. The latter is then never used (I can't find any `IP:` occurrence since the removal of rules related to IP reputation and DOS from the core).

Could it be reasonable to clean up both the variable and the `initcol` from the core and maybe add them to just some related plugins? It would avoid SHA1 computation (and storage) of the User-Agent for every request, regardless of whether it is used or not.

- Basically, I'm raising again https://github.com/coreruleset/coreruleset/issues/377 (back in the day I think that it was still used)
- Actually, not a bug, just a clean-up / little performance improvement.
- CRS version (e.g., v3.2.0): `v4.0/dev`.

Initial conversation with @airween: https://github.com/coreruleset/coreruleset/pull/3043#issuecomment-1363884977.

Many thanks! Happy to provide a PR if seems okay to clean them up

**Comment by airween** [comment]
A quick idea: we should add a condition to these rules (901318 and 901321), which checks a plus `TX` variable (eg. `tx.init_ip_coll`), which could be set up in crs-setup.conf.

**Comment by dune73** [comment]
I agree this is in need of a cleanup. 

There are pros and cons to @airweens proposal.

Con: CRS does not do sessions anymore, let's not initiate them at all.
Pro: Plugins may want to use sessions and if they do it's better CRS initializes them.

I'm leaning slightly on the CRS-should-init side. But with a config item like @airween proposes.

**Comment by fzipi** [comment]
Are we discussing this more broadly? Is this blocking for v4?

**Comment by dune73** [comment]
Depends on the solution we chose. :)

### Solution 1: It's a plugin problem, plugins need to deal with this

We rip out 901318 and 901321 and we're done.

### Solution 2: We make the initialization configurable, off by default, so plugin do not need to coordinate.

V4 depends on this minor change then.

Let's take a decision in the chat tonight.
